administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -367,301 +383,409 @@ public static function resizeImage(&$debugoutput, $src_file, $dest_file, $usefor
       return true;
     }
 
-    // For free resizing and cropping the center
-    $offsetx = null;
-    $offsety = null;
-    if($max_width)
-    {
-      $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_RESIZE_TO_MAX').'<br />';
-
-      if($new_width <= 0)
+    // determine resizing width and height
+      $offsetx = null;  
+      $offsety = null;
+      $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_CREATE_THUMBNAIL_FROM').' '.$imginfo['type'];
+      if ($anim && $imginfo['animation'])
       {
-        $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_ERROR_NO_VALID_WIDTH_OR_HEIGHT').'<br />';
-
-        return false;
+        $debugoutput .= ' '.JText::_('COM_JOOMGALLERY_UPLOAD_CREATE_WITH_ANIM');
       }
+      elseif ($imginfo['transparency'])
+      {
+        $debugoutput .= ' '.JText::_('COM_JOOMGALLERY_UPLOAD_CREATE_WITH_TRANS');
+      }
+      elseif ($anim && $imginfo['animation'] && $imginfo['transparency'])
+      {
+        $debugoutput .= ' '.JText::_('COM_JOOMGALLERY_UPLOAD_CREATE_WITH_TRANS_ANIM');
+      }
+      $debugoutput .= ', '.$imginfo['width'].' x '.$imginfo['height'].'...<br />';
 
-      $ratio = max($srcHeight,$srcWidth) / $new_width ;
-    }
-    else
-    {
-      // Resizing to thumbnail
-      $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_CREATE_THUMBNAIL_FROM').' '.$imginfo[2].', '.$imginfo[0].' x '.$imginfo[1].'...<br />';
-
-      if($new_width <= 0 || $thumbheight <= 0)
+      if($new_width <= 0 || $new_height <= 0)
       {
         $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_ERROR_NO_VALID_WIDTH_OR_HEIGHT').'<br />';
 
         return false;
       }
 
-      switch($useforresizedirection)
+      switch($settings)
       {
-        // Convert to height ratio
```

Comment:
case 0 and above ratio calculation is replaced by case 0, case 1 and case 2. It results the same behaviour but the code is much cleaner and easier to understand. With this change the function arguments become much more intuitive.

case 0: resize image to heigh ratio
case 1: resize image to width ratio
case 2: resize image to wether height or width ratio depending on longer side (width or height)